src/components/tab-bar/tab.types.ts

```diff
@@ -0,0 +1,26 @@
+export interface Tab {
+    /**
+     * Name of the tab.
+     */
+    name: string;
+
+    /**
+     * Name of the icon to use.
+     */
+    icon?: string;
+
+    /**
+     * True if the tab should be selected.
+     */
+    selected?: boolean;
+
+    /**
+     * A text shown when a tab is selected.
+     */
+    content?: string;
```

Comment:
This should not be part of the actual tab, but instead be provided by a `<slot>` element